### DISCUSSION TITLE
Fetch the data on the client

### DISCUSSION THREAD

**Discussion by NoeReverse** [posva/pinia-colada#123] [Questions] [answered]
Hello! I noticed that Pinia Colada by default executes all queries only on the server, which is very good because Vue Query [requires the use of `suspense` function](https://tanstack.com/query/v5/docs/framework/vue/guides/ssr) to execute queries this way. It's really inconvenient. With ssr, unless you use `suspense`, Vue Query will execute queries twice: once on the server, once on the client. So I like your approach of not requiring additional actions to run queries only on the server.

But sometimes we may want to execute a query only on the client. I suppose we could use the `enabled` parameter, like this:
```TypeScript
useQuery({
  key: ['todos'],
  query: () => fetch('/api/todos').then((res) => res.json())

  enabled: import.meta.client // ← for client only request
})
```
Is it correct?

In Nuxt's `useFetch`, you can simply pass the server option to achieve the same result. So maybe it makes sense to make the same option in Pinia Colada?

**Comment by posva** [comment, marked as answer]
Yes! Using enabled is the correct way 🙌
It’s worth being documented but I don’t think it’s worth having an alias unless we find some specific case not covered by enabled
